contracts/proxy_contract/v1/ergoscript/phoenix_v1_hodlcoin_proxy.es

```diff
@@ -1,39 +1,94 @@
 {
-
-    // ===== Contract Information ===== //
+    // ===== Contract Description ===== //
     // Name: Phoenix HodlCoin Proxy
     // Description: Contract guarding the proxy box for the HodlCoin protocol.
     // Version: 1.0.0
-    // Author: Luca D'Angelo (ldgaetano@protonmail.com), MGPai
+    // Author: mgpai22@github.com
 
     // ===== Box Contents ===== //
+
     // Tokens
-    // 1. (HodlCoinTokenId, HodlCoinTokenAmount)
+    // 1. HodlERGTokenId ? Only there if buyer is burning
+
     // Registers
-    // R4: SigmaProp    UserPK
-    // R5: Long         MinBoxValue
-    // R6: Long         MinerFee
-    // R7: Coll[Byte]   BankSingletonTokenId
-
-    // ===== Relevant Transactions ===== //
-    // 1. Mint Tx
-    // Inputs: Bank, Proxy
-    // DataInputs: None
-    // Outputs: Bank, UserPK, MinerFee, TxOperatorFee
-    // Context Variables: None
-    // 2. Burn Tx
-    // Inputs: Bank, Proxy
-    // DataInputs: None
-    // Outputs: Bank, UserPK, PhoenixFee, MinerFee, TxOperatorFee
-    // Context Variables: None
-
-    // ===== Compile Time Constants ($) ===== //
-    // None
+    // R4: SigmaProp => Buyer SigmaProp
+    // R5: Long => Miner Fee
+    // R6: Long => Minimum Erg Required to create a box
+
+
+    // ===== Compile Time Constants ===== //
+    // _minerFee: Long // Does not use SELF.R5 because buyer may put too low of a fee
+    // _bankSingleton: Coll[Byte]
 
-    // ===== Context Variables (@) ===== //
+    // ===== Context Extension Variables ===== //
     // None
 
-    // ===== Relevant Variables ===== //
+    val buyerPK: SigmaProp = SELF.R4[SigmaProp].get
+    val validBankBox: Boolean = INPUTS(0).tokens.size > 0 && INPUTS(0).tokens(0)._1 == _bankSingleton
+
+    if (validBankBox) {
+
+        val validMintOrBurnTx: Boolean = {
```

Comment:
split this up in two conditions using val isBurnTx: Boolean = (SELF.tokens.size == 1) && (SELF.tokens(0)._1 == HodlCoinTokenId)